### DISCUSSION TITLE
resizing controls

### DISCUSSION THREAD

**Discussion by visualizor** [picoe/Eto#2588] [Ideas]
I couldn't find a property for controls that allow them to be resized as their containers resize. Something similar to this in winforms
https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.autosizemode?view=windowsdesktop-8.0
Let me know if I simply missed it. Otherwise a good feature to have for the controls, I think.

**Comment by cwensley** [comment]
Hey @visualizor,

If you want the Form/Dialog to automatically change size based on the contents, you can set the `Window.AutoSize` property to true.

All other controls already auto size to their contents, it is not like in WinForms where you set a specific size then tell it to grow.  If you want a control to autosize, simply don't set its size.

Hope this helps,
Curtis.

**Reply by visualizor** [reply]
Thanks Curtis
I meant controls resizing with the container. So the other way around what you said.

**Reply by cwensley** [reply]
They already also do that.. if you resize a form or a parent control all of its child controls will resize along with it.

**Reply by cwensley** [reply]
If you had a more specific example of what you are trying to accomplish, it might be a bit easier to show how things might be able to be done.

**Reply by visualizor** [reply]
![clip](https://github.com/picoe/Eto/assets/25107288/79741d25-2c87-4125-b7d7-115642a10c08)

when i resize, the textboxes are clipped off. anyway to make them shorten as window(Form) gets smaller?
this is 2 `TextBox` in a `DynamicLayout` in an `Expander`, which is the `Content` of the `Form`

**Reply by cwensley** [reply]
When you add the controls to the DynamicLayout, set the xscale = true.  E.g.

```
var layout = new DynamicLayout();
layout.Add(new TextBox(), xscale: true);
layout.Add(new TextBox(), xscale: true);
```

**Reply by visualizor** [reply]
Turned out that I added them with `AddAutoSized()` 
Regular `Add()` would have done it, even without the `xscale` arg
My fault...

**Reply by visualizor** [reply]
On this note, what's your recommendation for proportionally scaling multiple columns/rows of controls during resizing? `DynamicLayout` seems to stretch the last column and bottom row only.

**Reply by cwensley** [reply]
The DynamicLayout (and TableLayout, which DynamicLayout uses), will scale any row/column with the corresponding xscale/yscale property set to true.  So if you want only the middle column to expand, but not the first/last, it'd be something like:

```
var layout = new DynamicLayout();
layout.BeginHorizontal();
layout.Add("Before");
layout.Add(new TextBox(), xscale: true);
layout.Add("After");
layout.EndHorizontal();

``` 

If no column/row has explicitly set xscale/yscale, then the last will be scaled, which is what you're seeing.

Hope this helps.

**Comment by visualizor** [comment]
problem solved by using the proper add method on `DynamicLayout`
